alica_engine/src/engine/blackboard/KeyMapping.cpp

```diff
@@ -0,0 +1,44 @@
+#include <algorithm>
+#include <alica_common_config/debug_output.h>
+#include <any>
+#include <assert.h>
+#include <engine/blackboard/Blackboard.h>
+#include <engine/blackboard/KeyMapping.h>
+#include <iostream>
+#include <sstream>
+
+namespace alica
+{
+
+void KeyMapping::setInput(const Blackboard* parent_bb, Blackboard* child_bb) const
+{
+    const auto lockedParentBb = LockedBlackboardRO(*parent_bb);
```

Comment:
I guess we are guaranteed no thread will be locking child blackboard when we are doing this (or setOutput)?